File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        Arrays.sort(points, new Comparator<int[]>() {
            
            public int compare(int[] a, int[] b) {
                return Integer.compare(a[0]*a[0] + a[1]*a[1], b[0]*b[0] + b[1]*b[1]);
                
            }
        });
        return Arrays.copyOfRange(points, 0, k);
    }
}

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
import heapq
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        minHeap=[]         # this is a list that is gonna be changed to heap by heapify keyword
        for x,y in points:
            dist= (x**2)+(y**2)
            minHeap.append([dist,x,y])  
        heapq.heapify(minHeap)
        
        final=[]
        while k>0:
            dist,x,y = heapq.heappop(minHeap)
            final.append(([x,y]))
            k -=1
        return final  
            
        
